tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4804 @@
+{
+  int ret = OB_SUCCESS;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+  } else if (OB_FAIL(ADD_DAG_WARN_INFO_PARAM(out_param, allocator, get_type(),
+                                             static_cast<int64_t>(OB_SERVER_TENANT_ID)))) {
+    STORAGE_LOG(WARN, "failed to fill info param", K(ret));
+  }
+  return ret;
+}
+int ObAdminFinishLogArchiveValidationDag::fill_dag_key(char *buf, const int64_t buf_len) const
+{
+  int ret = OB_SUCCESS;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_ISNULL(buf) || buf_len <= 0) {
+    ret = OB_INVALID_ARGUMENT;
+    STORAGE_LOG(WARN, "get invalid args", K(ret), KP(buf), K(buf_len));
+  } else {
+    if (OB_NOT_NULL(ctx_->log_archive_dest_)) {
+      if (OB_FAIL(databuff_printf(buf, buf_len, "log_archive_dest=%s",
+                                  ctx_->log_archive_dest_->get_root_path().ptr()))) {
+        STORAGE_LOG(WARN, "failed to fill dag key", K(ret));
+      }
+    } else {
+      if (OB_FAIL(databuff_printf(buf, buf_len, "backup_piece_path="))) {
+        STORAGE_LOG(WARN, "failed to fill dag key", K(ret));
+      } else {
+        FOREACH_X(path_iter, ctx_->backup_piece_path_array_, OB_SUCC(ret))
+        {
+          if (OB_NOT_NULL(path_iter)) {
+            if (OB_FAIL(databuff_printf(buf, buf_len, "%s%s ", buf,
+                                        (*path_iter)->get_root_path().ptr()))) {
+              STORAGE_LOG(WARN, "failed to fill dag key", K(ret));
+            }
+          } else {
+            ret = OB_ERR_UNEXPECTED;
+            STORAGE_LOG(WARN, "path_iter is null", K(ret));
+          }
+        }
+      }
+    }
+  }
+  return ret;
+}
+int64_t ObAdminFinishLogArchiveValidationDag::hash() const
+{
+  int64_t ptr = reinterpret_cast<int64_t>(this);
+  return common::murmurhash(&ptr, sizeof(ptr), 0);
+}
+int ObAdminFinishLogArchiveValidationDag::init(int64_t id, ObAdminBackupValidationCtx *ctx)
+{
+  int ret = OB_SUCCESS;
+  if (IS_INIT) {
+    ret = OB_INIT_TWICE;
+    STORAGE_LOG(WARN, "init twice", K(ret));
+  } else if (OB_ISNULL(ctx)) {
+    ret = OB_INVALID_ARGUMENT;
+    STORAGE_LOG(WARN, "ctx is null", K(ret));
+  } else {
+    is_inited_ = true;
+    id_ = id;
+    ctx_ = ctx;
+  }
+  return ret;
+}
+int ObAdminFinishLogArchiveValidationDag::create_first_task()
+{
+  int ret = OB_SUCCESS;
+  ObAdminFinishLogArchiveValidationTask *task = nullptr;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_FAIL(alloc_task(task))) {
+    STORAGE_LOG(WARN, "failed to alloc task", K(ret));
+  } else if (OB_FAIL(task->init(0 /*task_id, first mean*/, ctx_))) {
+    STORAGE_LOG(WARN, "failed to init task", K(ret));
+  } /*add task child here*/ else if (OB_FAIL(add_task(*task))) {
+    STORAGE_LOG(WARN, "failed to add task", K(ret));
+  } else {
+    STORAGE_LOG(INFO, "succeed to create first task", K(ret));
+  }
```

Comment:
We need handle exception if add task failed

Reply:
for task and dag schedule, the resource recovery willed a TODO